src/endpoint/s3/s3_utils.js

```diff
@@ -622,6 +622,28 @@ function encode_uri_unless_already_encoded(uri = '') {
     return is_uri_already_encoded(uri) ? uri : encodeURIComponent(uri);
 }
 
+/**
+ * parse_version_id throws an error if version_id is an empty string, and returns it otherwise
+ * @param {string|undefined} version_id
+ */
+function parse_version_id(version_id) {
+    if (version_id === '') {
+        throw new S3Error(S3Error.InvalidArgumentEmptyVersionId);
+    }
+    return version_id;
+}
+
+/**
+ * parse_version_id_marker throws an error if version_id_marker is an empty string, and returns it otherwise
+ * @param {string|undefined} version_id_marker
+ */
+function parse_version_id_marker(version_id_marker) {
+    if (version_id_marker === '') {
+        throw new S3Error(S3Error.InvalidArgumentEmptyVersionIdMarker);
+    }
+    return version_id_marker;
+}
+
```

Comment:
I don't think we need two functions, this is handling the exact same type of information - a version_id. Since only the error is different you can have a default argument for all cases and for that different case of version_id_marker to pass the desired error spec from the caller as S3Error.InvalidArgumentEmptyVersionIdMarker

```suggestion
/**
 * parse_version_id throws an error if version_id is an empty string, and returns it otherwise
 * @param {string|undefined} version_id
 * @param {import('./s3_errors').S3ErrorSpec} [empty_err]
 */
function parse_version_id(version_id, empty_err = S3Error.InvalidArgumentEmptyVersionId) {
    if (version_id === '') {
        throw new S3Error(empty_err);
    }
    return version_id;
}

```